specs/factorial_spec.rb

```diff
@@ -0,0 +1,15 @@
+require 'rspec'
+require_relative '../solver'
+
+RSpec.describe 'Solver' do
+  it 'Returns factorial of a positive integer' do
+    solver = Solver.new
+    factorial_result = solver.factorial(5)
+    expect(factorial_result).to eq(120)
+  end
+  it 'Returns factorial of a positive integer' do
+    solver = Solver.new
+    factorial_result = solver.factorial(4)
+    expect(factorial_result).to eq(24)
+  end
+end
```

Comment:
- [ ] Its a requirement that a test case should be present to test negative value which raises an exception as an expectation. Kindly add this.

e.g
```
describe Resolver do
         describe "#factorial" do
             it "raises exception for negative number" do
             resolver = Resolver.new
             expect { resolver.factorial(-5) }.to raise_exception
             end
         end
     end
```
screenshot of requirements check them out [here](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/testing/tdd_project.md)
![image](https://github.com/Yosefgeda/tdd-ruby/assets/81932478/46d37a65-0715-4fe5-82da-8967f42b687a)